exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/scan/TestFileScanFramework.java

```diff
@@ -25,6 +25,8 @@
 import java.util.Iterator;
 import java.util.List;
 
+import mockit.Mock;
+import mockit.MockUp;
```

Comment:
Please, if possible, try to avoid the use of Mocks here. These tend to work poorly in Eclipse. These tests try to use classes and interfaces designed to allow proper text fixture classes rather than dynamic mocks.

Reply:
These tests use non-existent data files for populating the implicit column values, but with new changes, the file system is used to obtain other implicit column values such as last modified time. So mock was used to avoid rewriting tests to use real files.

Reply:
Agree about mocks, please try to avoid them...

Reply:
What we want is to pass into the implicit column provider an interface that will answer file-specific questions. That is the thing that holds the file system. For testing, implement the interface to pretend to read files. For DFS, implement a real interface.

This will also allow someone to implement their own version for systems not based on DFS.